health/health.go

```diff
@@ -63,30 +72,36 @@ func check(
 	}()
 
 	start := time.Now()
+
+	onTick := func() {
+		if healthCheckFunc() {
+			start = time.Now()
+			if !healthy {
+				healthy = true
+				updates <- true
+			}
+		} else if healthy && time.Now().After(start.Add(healthTimeout)) {
+			healthy = false
+			updates <- false
+		}
+	}
+
 	ticker := time.NewTicker(pollDelay)
 	defer ticker.Stop()
 
+	onTick()
```

Comment:
Why should this fire immediately; isn't it ok to just wait for the first tick?

Reply:
It makes testing faster, otherwise we'd have to wait for pollDelay amount of time every time we swap out the backends. If doing this makes something else worse, then I guess we can just live with slower tests.